### PR TITLE
Box model for vumeter

### PR DESCRIPTION
**Pull request by jclaveau**
This add support of QSS Box model to the VuMeter widget.
Presently, some widget draw the borders, margins and paddings but the widget's content is not drawn in the good rectangle. After digging in the qstylesheetstyle class, I found a way to get those values : 

>    QRect contentRect = style()->subElementRect(QStyle::SE_FrameContents, &option, this);
>    if (contentRect.isNull()) {
>        contentRect = rect();
>    }

An example with the following QSS rules below : 

>  margin            : 9px 0 0 4px;
>  padding           : 2px 5px 10px 0px;
>  border-top        : 1px solid #ffffff;
>  border-bottom     : 1px dashed #ffffff;
>  background-color  : #ff0000;

mixxx_boxmodel_vumeter

### CODE REVIEW

**Comment by jclaveau** [comment]
As mentioned here https://github.com/mixxxdj/mixxx/pull/613, 
I created a fake skin to check the behaviour of different widgets : 
https://github.com/demos/developer_skins/tree/render_tests

File: src/widget/wvumeter.cpp [hunk 1]
```diff
@@ -165,24 +166,31 @@ void WVuMeter::paintEvent(QPaintEvent *) {
...
     option.initFrom(this);
     QStylePainter p(this);
+    p.setBrush(option.palette.text());
     p.drawPrimitive(QStyle::PE_Widget, option);
-
+    
+    QRect contentRect = style()->subElementRect(QStyle::SE_FrameContents, &option, this);
+    if (contentRect.isNull()) {
+        contentRect = rect();
+    }
```
Comment (daschuer) [inline src/widget/wvumeter.cpp]: Please add a comment why this call and check is required.

File: src/widget/wvumeter.cpp [hunk 2]
```diff
@@ -165,24 +166,31 @@ void WVuMeter::paintEvent(QPaintEvent *) {
...
+    
     if (!m_pPixmapBack.isNull() && !m_pPixmapBack->isNull()) {
         // Draw background. DrawMode takes care of whether to stretch or not.
-        m_pPixmapBack->draw(rect(), &p);
+        m_pPixmapBack->draw(contentRect, &p);
     }
-
+
```
Comment (daschuer) [inline src/widget/wvumeter.cpp]: Please configure your editor, that it removes tailing white space or at least not introduces some.

File: src/widget/wvumeter.cpp [hunk 3]
```diff
@@ -238,3 +248,33 @@ void WVuMeter::paintEvent(QPaintEvent *) {
...
 }
+
+QSize WVuMeter::sizeHint() const {
+    QStyleOption option;
+    option.initFrom(this);
+    
+    QSize widgetSize = WWidget::sizeHint();
+    
+    if (!m_pPixmapBack.isNull() && !m_pPixmapBack->isNull()) {
+        QSize backSize = style()->sizeFromContents(QStyle::CT_PushButton, &option,
```
Comment (daschuer) [inline src/widget/wvumeter.cpp]: Why QStyle::CT_PushButton? Please add a comment.

File: src/widget/wvumeter.cpp [hunk 4]
```diff
@@ -238,3 +248,33 @@ void WVuMeter::paintEvent(QPaintEvent *) {
     m_dLastParameter = m_dParameter;
     m_dLastPeakParameter = m_dPeakParameter;
 }
+
+QSize WVuMeter::sizeHint() const {
+    QStyleOption option;
+    option.initFrom(this);
+    
+    QSize widgetSize = WWidget::sizeHint();
```
Comment (daschuer) [inline src/widget/wvumeter.cpp]: What is the default WWidget::sizeHint()? 
What happens if the background is smaller than the VU Pixamp?

**Comment by jclaveau** [comment]
As written here https://github.com/mixxxdj/mixxx/pull/613, there is probably a way to enable boxmodel for every widget based on WWidget or QWidget (like for those based on QLabel) by using inheritance of QFrame so let's close this PR.

**Pull request closed without merging**